src/app/components/icons/utilityFunctions.tsx

```diff
@@ -15,21 +17,85 @@ export const getMuiIconName = (filename: string): string => {
     return muiName;
 };
 
-// Can be Material or PX Blue icons
-export const downloadSvg = (icon: IconType): void => {
-    if (icon.isMaterial) {
-        window.open(
-            `https://fonts.gstatic.com/s/i/materialicons/${icon.iconFontKey}/v6/24px.svg?download=true`,
-            '_blank'
+const getColorCode = (color:string): string => {
+    const lowerColor = color.toLowerCase();
+    switch(lowerColor){
+        case 'black':
+            return Colors.black[500];
+        case 'gray':
+            return Colors.gray[500];
+        case 'blue':
+            return Colors.blue[500];
+        case 'white':
+            return Colors.white[50];
+        default:
+            return Colors.black[500];
+    }
+}
+
+export const changeSvgColorAndSize = (svg: string, color: string | undefined, size: number | undefined): string => {
+    let newSvg = svg;
+    
+    if (color) {
+        newSvg = newSvg.replace(/<svg.*?>/i, (match) => match.replace('height=', `fill="${getColorCode(color)}" height=`));
+    }
+    if (size) {
+        newSvg = newSvg.replace(/<svg.*?>/i, (match) =>
+            match.replace(/height=".*?"/i, `height="${size}"`).replace(/width=".*?"/i, `width="${size}"`)
         );
-    } else {
-        window.open(`https://raw.githubusercontent.com/pxblue/icons/dev/svg/${getSnakeCase(icon.name)}.svg`, '_blank');
     }
+    return newSvg;
+};
+
+export const createDownloadSvgElement = (icon: IconType, iconData: string, color: string, size: number): void => {
+    // create a placeholder dom element and attach the new file download handler to it
+    const element = document.createElement('a');
+    element.setAttribute(
+        'href',
+        `data:text/plain;charset=utf-8, ${encodeURIComponent(changeSvgColorAndSize(iconData, color, size))}`
+        );
+    const iconName = [icon.name, color, '-', size, 'dp'].join('');
+    element.setAttribute('download', `${getSnakeCase(iconName).toLowerCase()}.svg`);
+    element.style.display = 'none';
+    document.body.appendChild(element);
+    element.click();
+    document.body.removeChild(element);
 };
 
-// Material Icons only
-export const downloadPng = (icon: IconType): void => {
-    window.open(`//fonts.gstatic.com/s/i/materialicons/${icon.iconFontKey}/v6/black-24dp.zip?download=true`, '_blank');
+export const createDownloadPxbPngElement = async (iconName: string, colorName: string, size: number): Promise<void> => {
+    const imageName = `${getSnakeCase(iconName)}_${colorName.toLocaleLowerCase()}_${size}dp.png`;
+    const imageSrc = `https://raw.githubusercontent.com/pxblue/icons/dev/png/png${size}/${imageName}`
+    const image = await fetch(imageSrc) || ''
+    const imageBlog = await image.blob() || ''
+    const imageURL = URL.createObjectURL(imageBlog)
+  
+    const link = document.createElement('a');
+    link.href = imageURL;
+    link.download = imageName;
+    document.body.appendChild(link);
+    link.click();
+    document.body.removeChild(link);
```

Comment:
Can we functionalize this element creation since the logic is the same for SVGs and PNGs